signer/src/eth.rs

```diff
@@ -31,35 +32,39 @@ impl Keypair {
     /// Create a keypair from a BIP-39 mnemonic phrase, optional password, account index, and
     /// derivation type.
     ///
+    /// **Note:** if the `std` feature is not enabled, we won't attempt to normalize the provided password
+    /// to NFKD first, and so this is your responsibility. This is not a concern if only ASCII
+    /// characters are used in the password.
+    ///
     /// # Example
     ///
     /// ```rust
     /// use subxt_signer::{ bip39::Mnemonic, eth::Keypair };
     ///
     /// let phrase = "bottom drive obey lake curtain smoke basket hold race lonely fit walk";
     /// let mnemonic = Mnemonic::parse(phrase).unwrap();
-    /// let keypair = Keypair::from_phrase(&mnemonic, None, 0, Default::default()).unwrap();
+    /// let keypair = Keypair::from_phrase(&mnemonic, None, DerivationPath::eth(0,0)).unwrap();
     ///
     /// keypair.sign(b"Hello world!");
     /// ```
     pub fn from_phrase(
         mnemonic: &bip39::Mnemonic,
         password: Option<&str>,
-        index: u32,
-        derivation: Derivation,
+        derivation_path: DerivationPath,
     ) -> Result<Self, Error> {
-        let derivation_path = match derivation {
-            Derivation::Hard => format!("m/44'/60'/{}'/0/0", index),
-            Derivation::Soft => format!("m/44'/60'/0'/0/{}", index),
-        }
-        .parse()
-        .map_err(Error::InvalidDerivationIndex)?;
-
-        let private = bip32::XPrv::derive_from_path(
-            mnemonic.to_seed(password.unwrap_or("")),
-            &derivation_path,
-        )
-        .expect("valid private key");
+        // `to_seed` isn't available unless std is enabled in bip39.
+        #[cfg(feature = "std")]
+        let seed = mnemonic.to_seed(password.unwrap_or(""));
+        #[cfg(not(feature = "std"))]
+        let seed = mnemonic.to_seed_normalized(password.unwrap_or(""));
+
+        // TODO: Currently, we use bip32 to derive private keys which under the hood uses
+        // the Rust k256 crate. We _also_ use the secp256k1 crate (which is very similar).
+        // It'd be great if we could 100% use just one of the two crypto libs. bip32 has
```

Comment:
let's open a PR to the bip32 crate to update the secp256k1 crate then :)

Reply:
I almost did this hehe, but yup definitely; I think when this merges we can open an issue in Subxt to avoid thedupe crates and then go from there :)